tests/test_csc.py

```diff
@@ -423,44 +416,97 @@ async def mount_offset_callback(self):
+                harness.pnematics.evt_m1State.set_put(state=ATPneumatics.AirValveState.OPENED)
+                harness.pnematics.evt_m2State.set_put(state=ATPneumatics.AirValveState.OPENED)
+
+                # set the hexapod callback
+                def hexapod_move_callback(data):
+                    harness.hexapod.evt_positionUpdate.put()
+
+                def m1_open_callback(data):
+                    harness.pnematics.evt_m1State.set_put(state=ATPneumatics.AirValveState.OPENED)
+
+                def m2_open_callback(data):
+                    harness.pnematics.evt_m2State.set_put(state=ATPneumatics.AirValveState.OPENED)
+
+                def m1_close_callback(data):
+                    harness.pnematics.evt_m1State.set_put(state=ATPneumatics.AirValveState.CLOSED)
+
+                def m2_close_callback(data):
+                    harness.pnematics.evt_m2State.set_put(state=ATPneumatics.AirValveState.CLOSED)
+
+                # Add callback to events
+                def callback(data):
+                    pass
+
+                # Set required callbacks
+                harness.hexapod.cmd_moveToPosition.callback = Mock(wraps=hexapod_move_callback)
+                harness.aos_remote.evt_detailedState.callback = Mock(wraps=callback)
+                harness.pnematics.cmd_m1SetPressure.callback = Mock(wraps=callback)
+                harness.pnematics.cmd_m2SetPressure.callback = Mock(wraps=callback)
+                harness.pnematics.cmd_m1OpenAirValve.callback = Mock(wraps=m1_open_callback)
+                harness.pnematics.cmd_m2OpenAirValve.callback = Mock(wraps=m2_open_callback)
+                harness.pnematics.cmd_m1CloseAirValve.callback = Mock(wraps=m1_close_callback)
+                harness.pnematics.cmd_m2CloseAirValve.callback = Mock(wraps=m2_close_callback)
+
+                # Give a telescope position
+                azimuth = 50.0
+                elevation = 45.0
+                await publish_mountEncoders(harness, azimuth, elevation, ntimes=1)
+                harness.atmcs.evt_allAxesInPosition.set_put(inPosition=True)
+
                 # if there is nothing (atpneumatics/atspectrograph) sending
                 # events then this command times out need to extend the
                 # timeout in this case.
+                harness.aos_remote.evt_correctionOffsets.flush()
+
                 await salobj.set_summary_state(harness.aos_remote, salobj.State.ENABLED, timeout=60)
                 self.assertEqual(harness.csc.summary_state, salobj.State.ENABLED)
 
-                offset_init = await harness.aos_remote.evt_correctionOffsets.next(
-                    flush=False,
-                    timeout=STD_TIMEOUT)
+                # Check that offset event was published on enable
+                offset_init = await harness.aos_remote.evt_correctionOffsets.aget(timeout=STD_TIMEOUT)
 
                 offset = {'m1': 1.0,
+                          'z': 1.0,
```

Comment:
Any reason for changing this order? Dictionaries in python are unordered.

Reply:
no idea... reverted to original.